src/main/java/com/google/sps/data/FirestoreDB.java

```diff
@@ -334,6 +397,26 @@ public Query recipesMatchingTags(Iterable<String> tagIds, Iterator<String> iter)
     return DBUtils.recipeMetadata();
   }
 
+  // todo: add to interface (returns list of recipeIDs matching any of the tags)
+  public List<RecipeMetadata> recipesMatchingAnyTags(List<String> tagIds) {
+    CollectionReference recipes = DBUtils.recipeMetadata();
+    Set<RecipeMetadata> metadata = new HashSet<RecipeMetadata>();
+    for (String tagId : tagIds) {
+      // Get only relevant recipes from the last week.
+      Calendar calendar = Calendar.getInstance();
+      calendar.add(Calendar.WEEK_OF_YEAR, -1);
```

Comment:
Does this work when it is currently the first week of the year?

Reply:
That's a good question -- I'll have to look into it, but I think it should